banyan/src/forest/read.rs

```diff
@@ -162,44 +183,39 @@ where
                         self.query
                             .intersecting(start_offset, index, &mut q_matching);
                         debug_assert_eq!(branch.children.len(), q_matching.len());
-                        let _ = std::mem::replace(matching, q_matching);
+                        head.filter.replace(q_matching);
 
                         if matches!(self.mode, Mode::Backward) {
```

Comment:
I think it would be clearer here to set the position in both cases, instead of using the default if Mode is Forward.